hail/src/main/scala/is/hail/io/fs/AzureStorageFS.scala

```diff
@@ -0,0 +1,388 @@
+        .build();
+      new AzureBlobServiceClientCache(clientSecretCredential)
+  }
+
+  def getBlobServiceClient(account: String): BlobServiceClient = {
+    storageClientCache.getServiceClient(account)
+  }
+
+  def getBlobClient(filename: String): BlobClient = {
+    val (account, container, path) = getAccountContainerPath(filename)
+    getBlobServiceClient(account).getBlobContainerClient(container).getBlobClient(path)
+  }
+
+  def getContainerClient(filename: String): BlobContainerClient = {
+    val (account, container, _) = getAccountContainerPath(filename)
+    getBlobServiceClient(account).getBlobContainerClient(container)
+  }
+
+  def openNoCompression(filename: String): SeekableDataInputStream = {
+    val blobClient: BlobClient = getBlobClient(filename)
+
+    val is: SeekableInputStream = new FSSeekableInputStream {
+      private[this] var client: BlobClient = blobClient
+
+      def fill(): Unit = {
+        bb.clear()
+
+        // read some bytes
+        val outputStream: ByteArrayOutputStream = new ByteArrayOutputStream()
+        val blobProperties = client.getProperties
+        val blobSize = blobProperties.getBlobSize
+
+        val count = Math.min(blobSize - pos, bb.remaining() - 1)
+        if (count <= 0) {
+          eof = true
+          return
+        }
+
+        // calculate the minimum of the bytes remaining in the buffer and the bytes remaining
+        // to be read in the blob
+        client.downloadWithResponse(
+          outputStream, new BlobRange(pos, count),
+          null, null, false, Duration.ofMinutes(1), null)
+        pos += count
+        bb.put(outputStream.toByteArray)
+        bb.flip()
+
+        assert(bb.position() == 0 && bb.remaining() > 0)
+      }
+
+      override def seek(newPos: Long): Unit = {
+        bb.clear()
+        bb.limit(0)
+        pos = newPos
+      }
+    }
+
+    new WrappedSeekableDataInputStream(is)
+  }
+
+  def createNoCompression(filename: String): PositionedDataOutputStream = {
+    val appendClient: AppendBlobClient = getBlobClient(filename).getAppendBlobClient
+    if (!appendClient.exists()) appendClient.create()
+
+    val os: PositionedOutputStream = new FSPositionedOutputStream {
+      private[this] val client: AppendBlobClient = appendClient
+
+      override def flush(): Unit = {
+        bb.flip()
+
+        if (bb.limit() > 0) {
+          var toWrite: mutable.MutableList[Byte] = mutable.MutableList()
+          var i = 0
+          while (i < bb.remaining()) {
+            toWrite += bb.get(i)
+            i += 1
+          }
+          client.appendBlock(new ByteArrayInputStream(toWrite.toArray), toWrite.length)
+        }
+
+        bb.clear()
+      }
+
+      override def close(): Unit = {
+        if (!closed) {
+          flush()
+          closed = true
+        }
+      }
+    }
+
+    new WrappedPositionedDataOutputStream(os)
+  }
+
+  def delete(filename: String, recursive: Boolean): Unit = {
+    if (this.exists(filename)) {
```

Comment:
I think this is being a little overprotective here. Errors can sometimes be a good thing! If I try to delete something that doesn't exist, I expect to get an error saying "Hey, this doesn't actually exist". Silently doing nothing is surprising behavior and could prevent me from realizing my mistake earlier.